lib/draw/BpmnRenderer.js

```diff
@@ -1978,15 +1975,15 @@ export default function BpmnRenderer(
         // conditional flow marker
         if (semantic.get('conditionExpression') && is(sourceSemantic, 'bpmn:Activity')) {
           svgAttr(connection, {
-            markerStart: marker('conditional-flow-marker', fill, stroke)
+            markerStart: marker(parentGfx, 'conditional-flow-marker', fill, stroke)
           });
         }
 
         // default marker
         if (sourceSemantic.get('default') && (is(sourceSemantic, 'bpmn:Gateway') || is(sourceSemantic, 'bpmn:Activity')) &&
             sourceSemantic.get('default') === semantic) {
           svgAttr(connection, {
-            markerStart: marker('conditional-default-flow-marker', fill, stroke)
+            markerStart: marker(parentGfx, 'conditional-default-flow-marker', fill, stroke)
```

Comment:
In a nutshell this renders markers locally, so as a result they can be styled with the element? We compromise by having a slightly bigger DOM tree, but I think that is fine.

Reply:
Correct. We do not de-duplicate markers by type and color anymore. This allows each marker to be styles individually